ppts/views.py

```diff
@@ -1,6 +1,77 @@
 from django.shortcuts import render
 from django.http import HttpResponse
+from django.http import Http404
+
+import matplotlib.pyplot as plt
+import PIL, PIL.Image
+import io
+from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
+
+from ppts.models import DwellingType
+from ppts.models import LandUse
+from ppts.models import Location
+from ppts.models import Planner
+from ppts.models import ProjectFeature
+from ppts.models import ProjectDescription
+from ppts.models import Record
+from ppts.models import RecordType
+from django.db.models import Count
 
 def index(request):
-    context = {}
-    return render(request, 'ppts/index.html', context)
\ No newline at end of file
+    graph_list = []
+    
+    #append names of graphs
+    graph_list.append('samplegraph')
+    graph_list.append('samplegraph2')
+    graph_list.append('samplegraph')
+    
+    #use list of graphs as context and render page
+    context = {'graph_list': graph_list}
+    return render(request, 'ppts/index.html', context)
+
+def graphmanager(request, graphname):
+    '''given the name of the graph, sends request to appropriate function'''
+    graph_func = globals()['graph_' + graphname]
```

Comment:
This is a bit odd, but I won't block this PR on it.

A more idiomatic solution would be to register each view separately with a specific URL path. This is more discoverable for new developers and avoids magic. Yes, it's more verbose, but verbosity is better than cleverness 9 times out of 10.

Reply:
I think reducing code redundancy is good at this point so that when we make changes we only need to make them in one spot.  We can change the structure later.